Composer/packages/extensions/visual-designer/src/widgets/ActionCard.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from 'react';
+
+import { FormCard } from '../components/nodes/templates/FormCard';
+import { WidgetContainerProps, WidgetComponent } from '../schema/uischema.types';
+import { ObiColors } from '../constants/ElementColors';
+import { getFriendlyName } from '../components/nodes/utils';
+
+export interface ActionCardProps extends WidgetContainerProps {
+  title: string;
+  icon: string;
+  content: string | number | JSX.Element;
+  menu: JSX.Element;
+  colors?: {
+    theme: string;
+    icon: string;
+  };
+  onClick: () => any;
+}
+
+const truncateType = $type => (typeof $type === 'string' ? $type.split('Microsoft.')[1] : '');
+
+const DefaultCardColor = {
+  theme: ObiColors.AzureGray3,
+  icon: ObiColors.AzureGray2,
+};
+
+export const ActionCard: WidgetComponent<ActionCardProps> = ({
+  data,
+  title,
+  icon,
+  content,
+  menu,
+  onClick,
+  colors = DefaultCardColor,
+}) => {
+  const header = getFriendlyName(data) || title || truncateType(data.$type);
```

Comment:
`title` is coming from the uischema right? Which is the output of `generateSDKTitle()`. I think for the sake of clarity, the title should just be passed through.

Another option is to move the call to `generateSDKTitle` into this and pass `title` into it from the ui schema.

Reply:
I prefer the solution that we invoking `getSDKTitle` in uischema directly
```ts
'title': data => getSDKTitle(data, 'Override title to this string')
```

Reply:
My prev comment is not correct. Now it's resolved as Andy described (move the call to `generateSDKTitile` into `ActionCard`)